aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,102 @@
+[!code-html[Main](memory/sample/WebCache/Views/Home/Index.cshtml)]
 
-Possible `eviction reasons` are:
+The cached `DateTime` value will remain in the cache while there are requests within the timeout period (and no eviction due to memory pressure). The image below shows the current time and an older time retrieved from cache:
 
-**None**
+![Index view with two different times displayed](memory/_static/time.png)
 
-No reason known.
+The following code uses [GetOrCreate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions#Microsoft_Extensions_Caching_Memory_CacheExtensions_GetOrCreate__1_Microsoft_Extensions_Caching_Memory_IMemoryCache_System_Object_System_Func_Microsoft_Extensions_Caching_Memory_ICacheEntry___0__) and [GetOrCreateAsync](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions#Microsoft_Extensions_Caching_Memory_CacheExtensions_GetOrCreateAsync__1_Microsoft_Extensions_Caching_Memory_IMemoryCache_System_Object_System_Func_Microsoft_Extensions_Caching_Memory_ICacheEntry_System_Threading_Tasks_Task___0___) to cache data. 
 
-**Removed**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet2&highlight=3-7,14-19)]
 
-The item was manually removed by a call to `Remove()`
+The following code calls [Get](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to fetch the cached time:
 
-**Replaced**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_gct)]
 
-The item was overwritten.
+See [IMemoryCache methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) and [CacheExtensions methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) for a description of the cache methods.
 
-**Expired**
+## Using MemoryCacheEntryOptions
 
-The item timed out.
+The following sample:
 
-**TokenExpired**
+- Sets the absolute expiration time. This is the maximum time the entry can be cached and prevents the item from becoming too stale when the sliding expiration is continuously renewed.
+- Sets a sliding expiration time. Requests that access this cached item will reset the sliding expiration clock.
+- Sets the cache priority to `CacheItemPriority.NeverRemove`. 
+- Sets a [PostEvictionDelegate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.postevictiondelegate) that will be called after the entry is evicted from the cache. The callback is run on a different thread from the code that removes the item from the cache.
 
-The token the item depended upon fired an event.
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_et)]
```

Comment:
The sample here looks very busy. If we want to illustrate the usage of the expiration settings and eviction callback `EvictionTime` and `CheckEvictionTime` aren't necessary. However, I do agree that they are useful to show more context around how these options are being used and called. In that case, It might be a good idea to highlight the relevant calls in `GetMemCacheOptions`